vespalib/src/tests/datastore/unique_store/unique_store_test.cpp

```diff
@@ -291,4 +295,33 @@ TYPED_TEST(TestBase, store_can_be_saved)
 
 #pragma GCC diagnostic pop
 
+TEST_F(DoubleTest, nan_is_handled)
+{
+    std::vector<double> myvalues = {
+        std::numeric_limits<double>::quiet_NaN(),
+        std::numeric_limits<double>::infinity(),
+        -std::numeric_limits<double>::infinity(),
+        10.0,
+        -std::numeric_limits<double>::quiet_NaN(),
+        std::numeric_limits<double>::infinity(),
+        -std::numeric_limits<double>::infinity()
+    };
+    std::vector<EntryRef> refs;
+    refs.push_back(EntryRef());
+    for (auto &value : myvalues) {
+        refs.emplace_back(add(value));
+    }
+    trimHoldLists();
+    EXPECT_TRUE(std::isnan(store.get(refs[1])));
+    EXPECT_TRUE(std::signbit(store.get(refs[1])));
```

Comment:
Consider checking ref 2 (+inf) and ref 3 (-inf) as well as those are represented as different unique values.